beacon-chain/p2p/rpc_topic_mappings.go

```diff
@@ -54,3 +96,94 @@ func VerifyTopicMapping(topic string, msg interface{}) error {
 	}
 	return nil
 }
+
+// TopicDeconstructor splits the provided topic to its logical sub-sections.
+// It is assumed all input topics will follow the specific schema:
+// /protocol-prefix/message-name/schema-version/...
+// For the purposes of deconstruction, only the first 3 components are
+// relevant.
+func TopicDeconstructor(topic string) (string, string, string, error) {
```

Comment:
this doesn't need to be exported in your current design

Reply:
I initially was working on the Altair branch, but changed it to develop since I could try getting non v2 changes into here. This is why this method is exported right now.

Reply:
Fair. Altair is just around the corner. Hopefully someone's code inspect doesn't complain about this before then